namespaces/mson-namespace.md

```diff
@@ -0,0 +1,503 @@
+
+Note not every MSON _Base Type_ is presented in JSON namespace primitive types and vice versa, see the table bellow:
+
+### JSON Namespace vs. MSON built-in types
+
+| JSON primitive |   JSON Namespace   | MSON Base Type |   MSON Namespace   |
+|:--------------:|:------------------:|:--------------:|:------------------:|
+|     boolean    |  JSON:Boolean Type |     boolean    |  MSON:Boolean Type |
+|     string     |  JSON:String Type  |     string     |  MSON:String Type  |
+|     number     |  JSON:Number Type  |     number     |  MSON:Number Type  |
+|      array     |   JSON:Array Type  |      array     |   MSON:Array Type  |
+|        -       |          -         |      enum      |   MSON:Enum Type   |
+|     object     |  JSON:Object Type  |     object     |  MSON:Object Type  |
+|      null      |   JSON:Null Type   |        -       |          -         |
+|        -       | JSON:Property Type |        -       | MSON:Property Type |
+
+
+# General Elements Definition
+
+General elements defined inside MSON namespaces but possibly reusable in another domain.
+
+## Identifier (Enum)
+
+Identifier of an element in the MSON namespace. The identifier MUST be either directly a `string` value or an `Id Element` element.
+
+### Members
+
+- (string)
+- (Id Element)
+
+## Id Element (Distinct Element)
+
+Element representing an identifier in the MSON namespace.
+
+### Properties
+
+- `element`: id (string, required, fixed)
+- `content` (string, required)
+
+## Distinct Element (Element)
+
+Abstract element representing an element that can be referenced or is a reference to another element.
+
+### Properties
+
+- `element` (Identifier, required) - identifier of this element's base element
+- `attributes`
+    - `id` (Identifier) - identifier of this element
+    - `ref` (Identifier) - reference to expanded element
+
+### Example
+
+```json
+{
+    "element": "number",
+    "attributes": {
+        "id": "my-number"
+    },
+    "content": 42
+}
+```
+
+which is identical to:
+
+```json
+{
+    "element": {
+        "element": "id",
+        "content": "number"
+    },
+    "attributes": {
+        "id": "my-number"
+    },
+    "content": 42
+}
+```
+
+and can be referred to as a _base element_:
+
+```json
+{
+    "element": "my-number"
+}
+```
+
+which _expands_ to:
+
+```json
+{
+    "element": "number",
+    "attributes": {
+        "ref": "my-number"
+    },
+    "content": 42
+}
+```
+
+## Select (Distinct Element)
+
+Element representing selection of options. Every item of content array represents one possible option.
+
+### Properties
+
+- `element`: select (string, required, fixed)
+- `content` (array[Option])
+
+## Option (Distinct Element)
```

Comment:
> Couldn't this just be implicit that the elements of the contents of a select element are the options?

No because there may be nested or grouped element, without `option` you couldn't tell whether its another "option" or a "group"

```html
<select>
  <option>
    <a></a>
    <p>A</p>
  </option>
  <option>
    <p>B</p>
  </option>
</select>
```

vs. 

```html
<select>
  <a></a>
  <p>A</p>
  <p>B</p>
</select>
```